src/dso_api/datasets/bagh/batch.py

```diff
@@ -0,0 +1,352 @@
+
+log = logging.getLogger(__name__)
+
+
+class ImportBagHTask(batch.BasicTask):
+    def __init__(self, *args, **kwargs):
+        self.table = f"{self.__class__.dataset}_{self.__class__.name}"
+        self.temp_table = f"{self.__class__.dataset}_temp"
+        self.path = kwargs.get("path")
+        self.models = kwargs["models"]
+        self.model = copy.deepcopy(self.models[self.__class__.name])
+        self.gob_path = kwargs.get("gob_path", "bag")
+        self.gob_id = {"bag": "BAG", "gebieden": "GBD"}[self.gob_path]
+
+        self.filename = f"{self.gob_id}_{self.__class__.name}_ActueelEnHistorie.csv"
+        self.source_path = f"{self.gob_path}/CSV_ActueelEnHistorie"
+        self.use_gemeentes = kwargs.get("use_gemeentes", False)
+        if self.use_gemeentes:
+            self.gemeentes = set()
+
+    def get_non_pk_fields(self):
+        return [x.name for x in self.model._meta.get_fields() if not x.primary_key]
+
+    def before(self):
+        cursor = connection.cursor()
+        cursor.execute(f"DROP TABLE IF EXISTS {self.temp_table}")
+        cursor.execute(
+            f"CREATE TEMPORARY TABLE {self.temp_table} AS TABLE {self.table} WITH NO DATA"
+        )
+        self.model._meta.db_table = self.temp_table
+
+        if self.path:
+            download_file(os.path.join(self.source_path, self.filename))
+
+        if self.use_gemeentes:
+            self.gemeentes = set(
+                self.models["gemeente"]
+                .objects.filter(eind_geldigheid__isnull=True)
+                .order_by("code")
+                .distinct("code")
+                .values_list("code", flat=True)
+            )
+
+    def after(self):
+        cursor = connection.cursor()
+        cursor.execute(f"ALTER TABLE {self.temp_table} ADD PRIMARY KEY(id)")
+        cursor.execute(f"CREATE INDEX ON {self.temp_table}(identificatie)")
+
+        if self.do_date_checks() > 0:
+            log.error(f"Data invalid. Skip table {self.table}")
+            return
+
+        # validate_geometry(models.Stadsdeel)
+
+        # Check rows to delete. In history database there should be no rows to delete
+        cursor.execute(
+            f"""
+            SELECT COUNT(e.*) FROM {self.table} e
+            LEFT JOIN  {self.temp_table} t ON e.id = t.id
+            WHERE t.id IS NULL
+            """
+        )
+        (count,) = cursor.fetchone()
+        if count > 0:
+            log.error(f"Rows deleted. Data invalid. Skip table {self.table}")
+            return
+        with transaction.atomic():
+            cursor.execute(
+                f"""
+                INSERT INTO  {self.table}
+                SELECT t.* FROM {self.temp_table} t
+                LEFT JOIN  {self.table} e ON t.id = e.id
+                WHERE e.id IS NULL
+                """
+            )
+            log.info(f"Inserted into {self.table} : {cursor.rowcount}")
+            setters = map(lambda x: f"{x} = t.{x}", self.get_non_pk_fields())
```

Comment:
Dit verdient wat commentaar dat we hier alleen velden overzetten, en dus geen SQL injectie mogelijk maken.